webapp/channels/src/components/login/login.tsx

```diff
@@ -430,9 +432,12 @@ const Login = ({onCustomizeHeader}: LoginProps) => {
                 // our session after we use it to complete the sign in change.
                 LocalStorageStore.setWasLoggedIn(false);
             } else {
+                setSessionExpired(true);
+                setDesktopAppSessionExpired(true);
+
                 // Although the authority remains the local sessionExpired bit on the state, set this
                 // extra field in the querystring to signal the desktop app.
-                setSessionExpired(true);
+                // This is legacy support for older Desktop Apps and can be removed eventually
                 const newSearchParam = new URLSearchParams(search);
                 newSearchParam.set('extra', Constants.SESSION_EXPIRED);
                 history.replace(`${pathname}?${newSearchParam}`);
```

Comment:
Any reason not to move this to the desktop api, so everything that is to be removed is in the same place?

Reply:
Nothing specifically, I guess I worry a bit about someone using the `setSessionExpired` call and getting this param attached unexpectedly. That's my only concern. 1/5 on keeping it here.